src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java

```diff
@@ -115,7 +120,10 @@ public void handle(RoutingContext context) {
                         addToEvent(result.getRight(), ampEventBuilder::bidResponse, result))
                 .map((Tuple2<BidRequest, BidResponse> result) -> toAmpResponse(result.getLeft(), result.getRight()))
                 .recover(this::updateErrorRequestsMetric)
-                .map(ampResponse -> addToEvent(ampResponse.getTargeting(), ampEventBuilder::targeting, ampResponse))
+                .compose(result ->
+                  ampResponsePostProcessor.postProcess(result.getLeft(), result.getMiddle(), result.getRight(),
+                           context.queryParams())
+                   .map(ampResponse -> addToEvent(ampResponse.getTargeting(), ampEventBuilder::targeting, ampResponse)))
```

Comment:
It's a small thing but would be more straightforward if this `map` operation was performed on result of `compose`, not on result of `ampResponsePostProcessor.postProcess` because this `map` call is a part of the top-level processing pipeline.